src/api.rs

```diff
@@ -466,19 +466,18 @@ impl Config {
 
     let pool = rayon::ThreadPoolBuilder::new().num_threads(threads).build().unwrap();
 
-    Context {
-      inner: ContextInner {
+    let inner = ContextInner {
         frame_count: 0,
         limit: 0,
         idx: 0,
         frames_processed: 0,
         frame_q: BTreeMap::new(),
         frame_invariants: BTreeMap::new(),
         keyframes: BTreeSet::new(),
+        keyframes_tmp: BTreeSet::new(),
```

Comment:
Please come up with a better name for this...

Reply:
The plan is to have only one of this in the end but it would be `detected_keyframes`. if it has to stay.

Reply:
What this really is is "first pass data". detected_keyframes works for now.